terraform/anomaly_detection/main.tf

```diff
@@ -0,0 +1,213 @@
+  iam_project_roles = {
+    (module.google_cloud_project.project_id) = [
+      "roles/storage.admin",
+      "roles/dataflow.worker",
+      "roles/monitoring.metricWriter",
+      "roles/pubsub.editor"
+    ]
+  }
+}
+
+
+// Network
+module "vpc_network" {
+  source     = "github.com/GoogleCloudPlatform/cloud-foundation-fabric//modules/net-vpc?ref=v32.0.0"
+  project_id = module.google_cloud_project.project_id
+  name       = "${var.network_prefix}-net"
+  subnets = [
+    {
+      ip_cidr_range         = "10.1.0.0/16"
+      name                  = "${var.network_prefix}-subnet"
+      region                = var.region
+      enable_private_access = true
+      secondary_ip_ranges = {
+        pods     = "10.16.0.0/14"
+        services = "10.20.0.0/24"
+      }
+    }
+  ]
+}
+
+module "firewall_rules" {
+  // Default rules for internal traffic + SSH access via IAP
+  source     = "github.com/GoogleCloudPlatform/cloud-foundation-fabric//modules/net-vpc-firewall?ref=v32.0.0"
+  project_id = module.google_cloud_project.project_id
+  network    = module.vpc_network.name
+  default_rules_config = {
+    admin_ranges = [
+      module.vpc_network.subnet_ips["${var.region}/${var.network_prefix}-subnet"],
+    ]
+  }
+  egress_rules = {
+    allow-egress-dataflow = {
+      deny        = false
+      description = "Dataflow firewall rule egress"
+      targets     = ["dataflow"]
+      rules       = [{ protocol = "tcp", ports = [12345, 12346] }]
+    }
+  }
+  ingress_rules = {
+    allow-ingress-dataflow = {
+      description = "Dataflow firewall rule ingress"
+      targets     = ["dataflow"]
+      rules       = [{ protocol = "tcp", ports = [12345, 12346] }]
+    }
+  }
+}
+module "regional_nat" {
+  // So we can get to Internet if necessary (from the Dataflow region)
+  source         = "github.com/GoogleCloudPlatform/cloud-foundation-fabric//modules/net-cloudnat?ref=v32.0.0"
+  project_id     = module.google_cloud_project.project_id
+  region         = var.region
+  name           = "${var.network_prefix}-nat"
+  router_network = module.vpc_network.self_link
+}
+
+resource "local_file" "variables_script" {
+  filename        = "${path.module}/../../pipelines/anomaly_detection/scripts/00_set_variables.sh"
+  file_permission = "0644"
+  content         = <<FILE
+# This file is generated by the Terraform code of this Solution Guide.
+# We recommend that you modify this file only through the Terraform deployment.
+export PROJECT=${module.google_cloud_project.project_id}
+export REGION=${var.region}
+export SUBNETWORK=regions/${var.region}/subnetworks/${var.network_prefix}-subnet
+export TEMP_LOCATION=gs://$PROJECT/tmp
+export SERVICE_ACCOUNT=${module.dataflow_sa.email}
+
+export DOCKER_REPOSITORY=${module.registry_docker.name}
+export IMAGE_NAME=dataflow-solutions-ml-ai
+export DOCKER_TAG=0.1
+export DOCKER_IMAGE=$REGION-docker.pkg.dev/$PROJECT/$DOCKER_REPOSITORY/$IMAGE_NAME
+
+export GCS_GEMMA_PATH=gs://$PROJECT/gemma_2B
```

Comment:
Remove unused option.

You will need to add other options too, for instance the details about the Bigtable instance and enrichment table.